hack/build-and-push.sh

```diff
@@ -133,95 +180,324 @@ if [ "$SKIP_BUILD" == "" ]; then
-oc kustomize --output "$generated_pipelines_dir" pipelines/
+GENERATED_PIPELINES_DIR=$(mktemp -d -p "$WORKDIR" pipelines.XXXXXXXX)
+declare -r GENERATED_PIPELINES_DIR
+oc kustomize --output "$GENERATED_PIPELINES_DIR" pipelines/
 
 # Generate YAML files separately since pipelines for core services have same .metadata.name.
-core_services_pipelines_dir=$(mktemp -d -p "$WORKDIR" core-services-pipelines.XXXXXXXX)
-oc kustomize --output "$core_services_pipelines_dir" pipelines/core-services/
+CORE_SERVICES_PIPELINES_DIR=$(mktemp -d -p "$WORKDIR" core-services-pipelines.XXXXXXXX)
+declare -r CORE_SERVICES_PIPELINES_DIR
+oc kustomize --output "$CORE_SERVICES_PIPELINES_DIR" pipelines/core-services/
 
-# Build tasks
-find task/*/*/ -maxdepth 0 -type d | awk -F '/' '{ print $0, $2, $3 }' | \
-while read -r task_dir task_name task_version
-do
-    if should_skip_repo "$QUAY_NAMESPACE" "task-$task_name"; then
-        echo "NOTE: not pushing task-$task_name:$task_version to $QUAY_NAMESPACE; the repo does not exist and $QUAY_NAMESPACE is deprecated"
-        continue
+
+inject_bundle_ref_to_pipelines() {
+    local -r task_name=$1
+    local -r task_version=$2
+    local -r task_bundle_with_digest=$3
+    sub_expr_1="
+        (.spec.tasks[].taskRef | select(.name == \"${task_name}\" and .version == \"${task_version}\" ))
+        |= {\"resolver\": \"bundles\", \"params\": [ { \"name\": \"name\", \"value\": \"${task_name}\" } , { \"name\": \"bundle\", \"value\": \"${task_bundle_with_digest}\" }, { \"name\": \"kind\", \"value\": \"task\" }] }
+    "
+    sub_expr_2="
+        (.spec.finally[].taskRef | select(.name == \"${task_name}\" and .version == \"${task_version}\" ))
+        |= {\"resolver\": \"bundles\", \"params\": [ { \"name\": \"name\", \"value\": \"${task_name}\" } , { \"name\": \"bundle\", \"value\": \"${task_bundle_with_digest}\" },{ \"name\": \"kind\", \"value\": \"task\" }] }
+    "
+    for filename in "$GENERATED_PIPELINES_DIR"/*.yaml "$CORE_SERVICES_PIPELINES_DIR"/*.yaml
+    do
+        yq e "$sub_expr_1" -i "${filename}"
+        yq e "$sub_expr_2" -i "${filename}"
+    done
+}
+
+# Get task version from task definition rather than the version in the directory path.
+# Arguments: task_file
+# The version is output to stdout.
+get_concrete_task_version() {
+    local -r task_file=$1
+    # Ensure an empty string is output rather than string "null" if the version label is not present
+    yq '.metadata.labels."app.kubernetes.io/version"' "$task_file" | sed '/null/d' | tr -d '[:space:]'
+}
+
+# Build and push a task as a Tekton bundle, that is tagged a floating tag and a fixed tag.
+# The task bundle reference with digest is output to stdout in the last line,
+# that is extracted from tkn-bundle-push.
+build_push_task() {
+    local -r task_dir=$1
+    local -r prepared_task_file=$2
+    local -r task_bundle=$3
+    local -r task_file_sha=$4
+    # local -r extra_annotations=${5:-""}
```

Comment:
If we don't need the `extra_annotations`, you can drop this commented code :slightly_smiling_face: